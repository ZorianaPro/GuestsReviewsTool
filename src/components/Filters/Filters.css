.Filters {
  display: block;
  width: 15em;
  position: fixed;
  margin-right: 1em;
  background: var(--colorWhite);
  padding-left: 1em;
  padding-right: 0.5em;
  border-radius: var(--borderRadius);
}

@media only screen and (max-width: 992px) {
  .Filters {
    width: 100%;
    position: relative;
    margin-bottom: 1em;
  }
}

.Filters button {
  display: none;
}

.Filters .Filters-Container {
  width: 100%;
  height: 100%;
}

.Filters .Filters-Category {

}

.Filters .Filters-Container .Filters-Category > h3 {
  text-transform: capitalize;
}

.Filters .Filters-Category .Filters-Category-Option {
  cursor: pointer;
  width: 100%;
  display: inline-block;
  font-size: var(--fontSize18);
}

.Filters .Filters-Container .Filters-Container-Content {
  display: block;
  position: relative;
}

.Filters .Filters-Category {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  font-size: var(--fontSize14);
  margin-bottom: 2.5em;
}

.Filters .Filters-Category .Filters-Category-Option label {
  margin-left: 0.5em;
}

@media only screen and (max-width: 992px) {
  .Filters .Forms-Container {
    display: flex;
  }
}