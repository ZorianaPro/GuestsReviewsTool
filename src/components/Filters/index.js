import React, { useCallback } from 'react';
import './Filters.css';

const Filters = ({
  filters,
  onChangeScore,
  onChangeChannel
}) => {
  const _onChange = useCallback((event) => {
    const option = event.target;
    const category = option.name;
    const form = document.querySelector(`form[name=${category}]`);
    const submitButton = form.querySelector('button[type="submit"]');
    submitButton.click();
  }, []);

  const onSubmit = useCallback((event) => {
    event.preventDefault();
    const category = event.target.name;
    if (category === 'score' && typeof onChangeScore === 'function') {
      onChangeScore(event);
    }
    if (category === 'channel' && typeof onChangeChannel === 'function') {
      onChangeChannel(event);
    }
  }, [onChangeChannel, onChangeScore]);

  return (
    <div className="Filters">
      <div className="Filters-Container">
        <div className="Filters-Container-Content">
          <h2>Filters</h2>
          <div className="Forms-Container">
            {
              filters
              && filters.map((filter) => (
                <form className="Filters-Category"
                  name={ filter.name }
                  onSubmit={ onSubmit }>
                  <h3>{ filter.name }</h3>
                  {
                    filter.values
                    && filter.values
                      .map((option, key) => (
                        <span className="Filters-Category-Option">
                          {
                            filter.name === 'channel'
                            && <input type="checkbox"
                              id={ `${filter.name}-${key}` }
                              key={ `${filter.name}-${key}` }
                              name={ filter.name }
                              value={ option.value }
                              onChange={ _onChange }
                            />

                          }
                          {
                            filter.name === 'score'
                            && <input type="radio"
                              id={ `${filter.name}-${key}` }
                              key={ `${filter.name}-${key}` }
                              name={ filter.name }
                              value={ option.value }
                              onChange={ _onChange }
                            />
                          }
                          <label htmlFor={ `${filter.name}-${key}` }>
                            { option.value }
                            {
                              filter.name === 'score'
                              && <span> & up</span>
                            }
                          </label>
                        </span>
                      ))
                  }
                  <button type="submit" />
                </form>
              ))
            }
          </div>
        </div>
      </div>
    </div>
  );
};

export default Filters;