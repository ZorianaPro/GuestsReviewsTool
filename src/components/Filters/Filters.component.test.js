import React from 'react';
import { shallow, mount } from 'enzyme';
import Filters from './';

describe('Filters', () => {
  const filtersArray = [
    {
      'name': 'score',
      'values': [
        {
          'value': 3
        },
        {
          'value': 2
        }
      ]
    },
    {
      'name': 'channel',
      'values': [
        {
          'value': 'AIRBNB'
        },
        {
          'value': 'SOMETHINGELSE'
        }
      ]
    }
  ];

  it('renders without crashing', () => {
    expect(
      shallow.bind(
        shallow,
        <Filters />
      ))
      .not
      .toThrow();
  });

  it('renders as expected', () => {
    expect(
      mount(
        <Filters />
      )
    ).toMatchSnapshot();
  });

  describe('on pass filters', () => {
    let filters;

    beforeAll(() => {
      filters = mount(
        <Filters
          filters={
            filtersArray
          }
        />
      );
    });

    it('renders correct amount of categories', () => {
      expect(filters.find('form.Filters-Category').length)
        .toEqual(filtersArray.length);
    });

    it('renders filter properties', () => {
      expect(filters.find('form.Filters-Category').first().find('input').length)
        .toEqual(2);
    });

    it('renders as expected', () => {
      expect(filters).toMatchSnapshot();
    });
  });
});
