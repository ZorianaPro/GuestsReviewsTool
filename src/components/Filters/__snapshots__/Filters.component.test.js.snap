// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters on pass filters renders as expected 1`] = `
<Filters
  filters={
    Array [
      Object {
        "name": "score",
        "values": Array [
          Object {
            "value": 3,
          },
          Object {
            "value": 2,
          },
        ],
      },
      Object {
        "name": "channel",
        "values": Array [
          Object {
            "value": "AIRBNB",
          },
          Object {
            "value": "SOMETHINGELSE",
          },
        ],
      },
    ]
  }
>
  <div
    className="Filters"
  >
    <h4>
      Filters
    </h4>
    <form
      className="Filters-Category"
      name="score"
      onSubmit={[Function]}
    >
      <p>
        score
      </p>
      <div>
        <input
          id="score-0"
          name="score"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          htmlFor="score-0"
        >
          3
          <span>
             & up
          </span>
        </label>
      </div>
      <div>
        <input
          id="score-1"
          name="score"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          htmlFor="score-1"
        >
          2
          <span>
             & up
          </span>
        </label>
      </div>
      <button
        type="submit"
      />
    </form>
    <form
      className="Filters-Category"
      name="channel"
      onSubmit={[Function]}
    >
      <p>
        channel
      </p>
      <div>
        <input
          id="channel-0"
          name="channel"
          onChange={[Function]}
          type="checkbox"
          value="AIRBNB"
        />
        <label
          htmlFor="channel-0"
        >
          AIRBNB
        </label>
      </div>
      <div>
        <input
          id="channel-1"
          name="channel"
          onChange={[Function]}
          type="checkbox"
          value="SOMETHINGELSE"
        />
        <label
          htmlFor="channel-1"
        >
          SOMETHINGELSE
        </label>
      </div>
      <button
        type="submit"
      />
    </form>
  </div>
</Filters>
`;

exports[`Filters renders as expected 1`] = `
<Filters>
  <div
    className="Filters"
  >
    <h4>
      Filters
    </h4>
  </div>
</Filters>
`;
